tests/domains_test.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+import scipp as sc
+from scipp.core.domains import find_domains
+import pytest
+
+x = sc.arange(dim='x', start=0, stop=10)
+
+
+def test_find_domains_fail_not_1d():
+    data = sc.ones(dims=['y', 'x'], shape=[2, 9], dtype='int64')
+    da = sc.DataArray(data=data, coords={'x': sc.concat([x, x], 'y')})
+    with pytest.raises(sc.DimensionError):
```

Comment:
It would be nice to have a more explicit error message than 'Cannot add coord exceeding DataArray dims'. Calling `find_domains` or `lookup` has, from a usage point, nothing to do with adding coords.

Reply:
Done